stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -33,3 +35,80 @@ def _compute_delivery_date_weekday(self):
                 channel.delivery_date_weekday = channel.shipment_date.weekday()
             else:
                 channel.delivery_date_weekday = -1
+
+    @property
+    def _delivery_date_generators(self):
+        d = super()._delivery_date_generators
+        d["customer"].append(self._next_delivery_date_partner_delivery_window)
+        return d
+
+    def _next_delivery_date_partner_delivery_window(self, delivery_date, partner):
+        """Get the next valid delivery date respecting customer delivery window.
+
+        The delivery date must be when the customer is open.
+        From the initial delivery_date, if the customer is not open on that
+        date and time, postpone to the start of the next open window.
+
+        A delivery date generator needs to provide the earliest valid date
+        starting from the received date. It can be called multiple times with a
+        new date to validate.
+        """
+        self.ensure_one()
+        partner.ensure_one()
+        if not self.respect_partner_delivery_time_windows:
+            while True:
+                delivery_date = yield delivery_date
+
+        if partner.delivery_time_preference == "anytime":
+            # no constrain, any date is valid
+            while True:
+                delivery_date = yield delivery_date
+
+        tz = partner.tz
+        if partner.delivery_time_preference == "workdays":
+            # postpone to Monday when date is on a week-end
+            while True:
+                delivery_date_tz = self._localize(delivery_date, tz=tz)
+                # postpone on Monday if Sat or Sun
+                if delivery_date_tz.weekday() < 5:  # Mon-Fri
+                    delivery_date = yield delivery_date
+                    continue
+                days = 0
+                if delivery_date_tz.weekday() == 5:  # Sat
+                    days = 1
+                elif delivery_date_tz.weekday() == 6:  # Sun
+                    days = 2
+                delivery_date_tz += timedelta(days=days)
```

Comment:
Better to use Odoo API for that:
```suggestion
                delivery_date_tz += fields.Datetime.add(delivery_date_tz, days=days)
```